cmake/modules/CTags.cmake

```diff
@@ -0,0 +1,21 @@
+find_program(CTAGS_EXECUTABLE ctags)
+if(CTAGS_EXECUTABLE)
+  set(src_dir "src")
+  set(tag_file ${src_dir}/tags)
+  execute_process(
+    COMMAND git config --file .gitmodules --get-regexp path
+    COMMAND awk "/${src_dir}/ { print $2 }"
+    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+    OUTPUT_VARIABLE submodules
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  string(REPLACE "\n" " " submodules ${submodules})
+  string(REPLACE "${src_dir}/" "--exclude=" excludes ${submodules})
+  separate_arguments(excludes UNIX_COMMAND ${excludes})
```

Comment:
I agree this is probably minimal, but it's extremely confusing even if you're aware of cmake's Lists and automagic shell-metacharacter quoting that includes spaces...so let's add a comment like this above it:

```
  # cmake will quote any embedded spaces when expanding ${excludes}
  # in the command below, so, turn ${excludes} into a cmake List (separated
  # by semicolons) to let the expansion below simply replace them with spaces
  # again.
```